packages/library/src/commonMain/kotlin/io/realm/internal/RealmObjectUtil.kt

```diff
@@ -56,15 +59,37 @@ fun RealmObjectInternal.unmanage() {
     this.`$realm$Pointer` = null
 }
 
-// Create a frozen copy of this object. Expected DB pointer is used by writes where we need to freeze
-// the object before the owner Realm is updated, but the pointer it will be updated with is already know.
-fun <T : RealmObject> RealmObjectInternal.freeze(realm: NativePointer, expectedDbPointer: NativePointer? = null): T {
+/**
+ * Creates a frozen copy of this object.
+ *
+ * @param frozenRealm Pointer to frozen Realm to which the frozen copy should belong.
+ */
+internal fun <T : RealmObject> RealmObjectInternal.freeze(frozenRealm: NativePointer): T {
```

Comment:
Shouldn't this be `T.freeze(...)`? But I guess you will then have to make an unchecked cast internally. Maybe we should include the type in `RealmObjectInternal<T>`?

Reply:
Playing around with the types, we end up being forced to do unchecked casts somewhere since the type system doesn't know that `RealmObjectInternal` is always implemented by a `RealmObject`. So for now it seems simpler to just keep the types as they are?